misk-testing/src/main/kotlin/misk/concurrent/FakeScheduledExecutorService.kt

```diff
@@ -17,21 +17,47 @@ import javax.inject.Singleton
 /**
  * ScheduledExecutorService for testing that runs in the current thread and is triggered using the
  * `tick()` method. An injected [Clock] is used to decide whether to execute a scheduled task.
+ *
+ * This service must always "catch up" to the clock, so fixed rate and fixed delay jobs are not different.
  */
 @Singleton
 class FakeScheduledExecutorService @Inject constructor(private val clock: Clock) :
     ScheduledExecutorService, ExecutorService by MoreExecutors.newDirectExecutorService() {
 
-  private val futures =
-      MultimapBuilder.treeKeys().arrayListValues().build<Long, RunnableFuture<*>>()
+  internal val futures = PriorityBlockingQueue<ScheduledFutureTask<*>>()
+
+  private var shutdown = false
 
   /** Check the current time on the clock and run any scheduled tasks that are due. */
   fun tick() {
-    futures.entries().stream()
-        .filter { entry -> entry.key <= clock.millis() }
-        .forEach { entry -> entry.value.run() }
+    var future = futures.poll()
+    while (future != null) {
+      if (this.isShutdown) {
+        future.cancel(false)
```

Comment:
Why did you need to add cancellation? If tick isn't called, no futures will run.

Reply:
Good point! Removed the cancellation.